libnavui-speedlimit/src/main/java/com/mapbox/navigation/ui/speedlimit/view/MapboxSpeedInfoView.kt

```diff
@@ -0,0 +1,447 @@
+
+            speedInfoPostedSpeedVienna.text = speedInfo.postedSpeed.toString()
+            speedInfoCurrentSpeedVienna.text = speedInfo.currentSpeed.toString()
+            speedInfoCurrentSpeedVienna.isVisible =
+                speedInfo.currentSpeed > speedInfo.postedSpeed && speedInfo.postedSpeed > 0
+        } else {
+            if (speedInfoOptions.showSpeedWhenUnavailable) {
+                showMutcdOrVienna(speedInfo.speedSignConvention)
+                speedInfoPostedSpeedMutcd.text = "--"
+                speedInfoCurrentSpeedMutcd.text = ""
+                speedInfoCurrentSpeedMutcd.isVisible = false
+
+                speedInfoPostedSpeedVienna.text = "--"
+                speedInfoCurrentSpeedVienna.text = ""
+                speedInfoCurrentSpeedVienna.isVisible = false
+            } else {
+                speedInfoMutcdLayout.isVisible = false
+                speedInfoViennaLayout.isVisible = false
+            }
+        }
+    }
+
+    /**
+     * Apply runtime changes to [MapboxSpeedLimitView] using [MapboxSpeedInfoOptions]
+     * @param speedInfoOptions
+     */
+    fun applyOptions(speedInfoOptions: MapboxSpeedInfoOptions) {
+        this.speedInfoOptions = speedInfoOptions
+        updateView()
+    }
+
+    private fun showMutcdOrVienna(speedSignConvention: SpeedLimitSign?) {
+        when (speedSignConvention) {
+            SpeedLimitSign.MUTCD -> {
+                speedInfoMutcdLayout.isVisible =
+                    speedInfoOptions.renderWithSpeedSign != SpeedLimitSign.VIENNA
+                speedInfoViennaLayout.isVisible =
+                    !speedInfoMutcdLayout.isVisible
+            }
+            SpeedLimitSign.VIENNA -> {
+                speedInfoViennaLayout.isVisible =
+                    speedInfoOptions.renderWithSpeedSign != SpeedLimitSign.MUTCD
+                speedInfoMutcdLayout.isVisible =
+                    !speedInfoViennaLayout.isVisible
+            }
+            // TODO: remove the null check after NN-305 is available
+            null -> {
+                when (speedInfoOptions.renderWithSpeedSign) {
+                    null -> {
+                        // no op
+                    }
+                    SpeedLimitSign.MUTCD -> {
+                        speedInfoMutcdLayout.isVisible = true
+                        speedInfoViennaLayout.isVisible = false
+                    }
+                    SpeedLimitSign.VIENNA -> {
+                        speedInfoMutcdLayout.isVisible = false
+                        speedInfoViennaLayout.isVisible = true
+                    }
+                }
+            }
+        }
+    }
+
+    private fun renderSpeedUnit(speedUnit: SpeedUnit) {
+        when (speedUnit) {
+            SpeedUnit.MILES_PER_HOUR -> {
+                speedInfoUnitTextMutcd.text = "mph"
+            }
+            SpeedUnit.KILOMETERS_PER_HOUR -> {
+                speedInfoUnitTextMutcd.text = "km/h"
+            }
```

Comment:
Should we translate these strings?

Reply:
I am not sure. @1ec5 Would you know anything about this, and if we need to have translations for units?

Reply:
@Zayankovsky I am gonna merge this and we can add the changes for translation in a follow up PR if required.